includes/class-wc-stripe-feature-flags.php

```diff
@@ -174,13 +182,24 @@ public static function are_apms_deprecated() {
 	 * @return bool
 	 */
 	public static function is_oc_available() {
-		$default_value   = self::get_option_with_default( self::OC_FEATURE_FLAG_NAME );
 		$stripe_settings = WC_Stripe_Helper::get_stripe_settings();
 		$pmc_enabled     = $stripe_settings['pmc_enabled'] ?? 'no';
+		if ( 'yes' !== $pmc_enabled ) {
+			return false;
+		}
+
+		/**
+		 * Filter to control the availability of the Optimized Checkout feature.
+		 *
+		 * @since 9.6.0
+		 * @deprecated This filter will be removed in version 9.9.0. No replacement will be provided as the Optimized Checkout feature will be permanently enabled.
+		 * @param string $default_value The default value for the feature flag.
+		 * @param string $pmc_enabled The value of the 'pmc_enabled' setting.
+		 */
 		return apply_filters(
 			'wc_stripe_is_optimized_checkout_available',
-			'yes' === $default_value && 'yes' === $pmc_enabled,
-			$default_value,
+			'yes' === $pmc_enabled,
```

Comment:
Given that we would have returned early above if `$pmc_enabled` is not `'yes'`, I think it would be clearer to default this to `true` instead.
```suggestion
			true,
```